theano/tensor/basic.py

```diff
@@ -4268,15 +4268,51 @@ def tile(x, reps, ndim=None):
     TODO: expand this.
     """
 
+    # if reps is tensor.scalar, integer or tensor.vector, we convert it to a list.
+    if not isinstance(reps, (list, tuple)):
+        reps_astensor = as_tensor_variable(reps)
+        ndim_check = reps_astensor.ndim
+
+        # tensor.scalar/integer case
+        if ndim_check == 0:
+            reps_ = []
+            reps_.append(reps)
+            reps = reps_
+        # tesor.vector case
+        elif ndim_check == 1:
+            if ndim == None:
+                raise ValueError("if reps is tensor.vector, you should specify "
+                                 "the ndim")
+            else:
+                offset = ndim-reps.shape[0]
+                # assert that reps.shape[0] does not exceed ndim
+                offset = theano.tensor.opt.assert_(offset, ge(offset, 0))
+                reps_ = [switch(i<offset, 1, reps[i-offset]) for i in range(ndim)]
+                reps = reps_
+        #other raise error
+        else:
+            raise ValueError("the dimension of reps should not exceed 1")
+    else:
+       ndim_check = None
+
+    # if reps.ndim is less than x.ndim, we pad the reps with
+    # "1" so that reps will have the same ndim as x. 
+    if len(reps) < x.ndim:
+       reps = [1]*(x.ndim-len(reps)) + reps
+
     try:
         iter(reps)
     except TypeError:
         raise ValueError("reps must be iterable")
```

Comment:
You can remove that check. At this point, you've made sure that `reps` is either a list or a tuple.